modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -533,16 +572,18 @@ public boolean equals(Object obj) {
         return true;
     }
 
-    abstract protected T getT(int index);
-    abstract protected int getIndex(T t);
+    protected abstract T getT(int index);
+    protected abstract int getIndex(T t);
 
-    /*
-     * Try to cast the arg to a T.
-     * @throws ClassCastException if the class of the argument is
-     *         is not a T
-     * @throws NullPointerException if the argument is null
+    /**
+     * Return {@code obj} if it is an instance of type {@code T},
+     * otherwise return {@code null}.
+     *
+     * @param obj the object to cast, cannot be {@code null}
+     * @return a type T, or {@code null} if the argument was not of this type
+     * @throws NullPointerException when {@code obj} is {@code null}
      */
-    abstract protected T cast(Object o);
+    protected abstract T cast(Object obj);
 
     protected long[] getBits() {
```

Comment:
Sure, I don't mind, the class is not public though, nor are its subclasses.

Reply:
I've made this method package private instead of protected.  The only reason it was protected was so that `toArray` implementations in the two subclasses could access it.  However, those implementations are totally unnecessary as `AbstractCollection` provides them, without bugs and likely with perfectly acceptable performance.

By basing `BitSet` on `AbstractSet`, which it probably should have been from the beginning, I also inherit contract respecting equals/hashCode implementations -- the only concession I did here is to provide a fast(er?) path when both sets are `BitSet`s.